src/modules/vtol_att_control/standard.cpp

```diff
@@ -283,51 +266,44 @@ void Standard::update_transition_state()
 	// copy virtual attitude setpoint to real attitude setpoint
 	memcpy(_v_att_sp, _mc_virtual_att_sp, sizeof(vehicle_attitude_setpoint_s));
 
+	const float trans_elapsed = hrt_elapsed_time(&_vtol_schedule.transition_start) / 1e6f;
+
 	if (_vtol_schedule.flight_mode == TRANSITION_TO_FW) {
 		if (_params_standard.front_trans_dur <= 0.0f) {
 			// just set the final target throttle value
 			_pusher_throttle = _params_standard.pusher_trans;
 
 		} else if (_pusher_throttle <= _params_standard.pusher_trans) {
+			const float progress = math::constrain(trans_elapsed / _params_standard.front_trans_dur, 0.0f, 1.0f);
+
 			// ramp up throttle to the target throttle value
-			_pusher_throttle = _params_standard.pusher_trans *
-					   (float)hrt_elapsed_time(&_vtol_schedule.transition_start) / (_params_standard.front_trans_dur * 1000000.0f);
+			_pusher_throttle = _params_standard.pusher_trans * progress;
 		}
 
+		// disable airbrakes output
+		_airbrakes_output = 0.0f;
+
+		const float as_blend_region = _params_standard.airspeed_trans - _params_standard.airspeed_blend;
+		const float as_above_blend = _airspeed->indicated_airspeed_m_s >= _params_standard.airspeed_blend;
```

Comment:
Here should be a negative sign instead of `>=`. Otherwise `_mc_throttle_weight` is a constant value.

Reply:
thanks, good catch